panther_power_control/src/power_board_node.py

```diff
@@ -171,54 +173,52 @@ def _motor_controllers_state_cb(self, msg: DriverState) -> None:
                 {msg.rear.fault_flag.can_net_err, msg.front.fault_flag.can_net_err}
             )
 
-    def _gpio_interrupt_cb(self, pin: int) -> None:
-        with self._pins_lock:
-            if pin == self._pins.SHDN_INIT:
-                self._chrg_sense_interrupt_time = rospy.get_time()
-
-            if pin == self._pins.E_STOP_RESET:
-                self._e_stop_interrupt_time = rospy.get_time()
-
     def _publish_pin_state_cb(self, *args) -> None:
         with self._pins_lock:
-            charger_state = self._read_pin(self._pins.CHRG_SENSE)
+            charger_state = self._lines['CHRG_SENSE'].get_value()
             self._publish_io_state('charger_connected', charger_state)
 
             # filter short spikes of voltage on GPIO
-            if rospy.get_time() - self._chrg_sense_interrupt_time > self._gpio_wait:
-                if self._read_pin(self._pins.SHDN_INIT):
-                    rospy.loginfo(f'[{rospy.get_name()}] Shutdown button pressed.')
-                    self._publish_io_state('power_button', True)
-                self._chrg_sense_interrupt_time = float('inf')
-
-            if rospy.get_time() - self._e_stop_interrupt_time > self._gpio_wait:
-                self._e_stop_event()
-                self._e_stop_interrupt_time = float('inf')
+            if self._lines['SHDN_INIT'].get_value():
+                if rospy.get_time() - self._chrg_sense_interrupt_time > self._gpio_wait:
+                    if self._lines['SHDN_INIT'].get_value():
+                        rospy.loginfo(f'[{rospy.get_name()}] Shutdown button pressed.')
+                        self._publish_io_state('power_button', True)
+                    self._chrg_sense_interrupt_time = float('inf')
+                elif math.isinf(self._chrg_sense_interrupt_time):
+                    self._chrg_sense_interrupt_time = rospy.get_time()
+
+            if self._lines['E_STOP_RESET'].get_value() != self._last_e_stop_state:
+                if rospy.get_time() - self._e_stop_interrupt_time > self._gpio_wait:
+                    self._e_stop_event()
```

Comment:
The first condition is not necessary as it is checked in `_e_stop_event()` method